### PR TITLE
Improve output of register and deregister operations

### PR DESCRIPTION
**Pull request by fschueller**
https://trello.com/c/IpPTOJJb/536-scpcnt-update-the-user-about-registration-and-de-registration-progress

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/158692/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 6060d67a4b6f5cd1b055d3c7ddb27b77b143165a into 529b91ef340e31070676604ed99b312a588dfedb)

**Comment by wstephenson** [comment]
The new coloured output methods in logger.rb need specs.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/158893/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 6a5cb3df7b954eed2efc50973d8761edb09af4c3 into 529b91ef340e31070676604ed99b312a588dfedb)

File: lib/suse/connect/client.rb [hunk 1]
```diff
@@ -240,17 +252,20 @@ def registered?
...
+          log.info '-> Removing service from system ...'
           System.remove_service service
         end
       end
 
       def print_success_message(product, action: 'Registered')
-        log.info "#{action} #{product.identifier} #{product.version} #{product.arch}"
+        log.info "\n#{action} #{product.identifier} #{product.version} #{product.arch}".bold
         log.info "Rooted at: #{@config.filesystem_root}" if @config.filesystem_root
         log.info "To server: #{@config.url}" if @config.url
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: "Deregistered *to* server" reads wrong, I'd use *from* in this case.  

I also think passing the verb as a string parameter to the message is going to be problematic when we have to translate SUSEConnect output, so perhaps just pass the action as a symbol and construct the output strings entirely within the function?

File: lib/suse/connect/client.rb [hunk 2]
```diff
@@ -83,6 +90,7 @@ def flatten_tree(tree)
       #
       # @returns: [Array] login, password tuple. Those credentials are given by SCC/Registration Proxy
       def announce_system(distro_target = nil, instance_data_file = nil)
+        log.info "Announcing system to SCC/RMT/SMT...\n".bold
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: Was 'SCC/RMT/SMT' blessed? Would 'registration server' be clearer to an outsider who is not familar with all of our registration proxies?

File: lib/suse/connect/client.rb [hunk 3]
```diff
@@ -94,6 +102,7 @@ def announce_system(distro_target = nil, instance_data_file = nil)
       # Re-send the system's hardware details on SCC
       #
       def update_system(distro_target = nil, instance_data_file = nil)
+        log.info 'Update system details on SCC/RMT ...\n'.bold
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: Should be consistent with the above.

File: lib/suse/connect/client.rb [hunk 4]
```diff
@@ -240,17 +252,20 @@ def registered?
...
         end
       end
 
       def print_success_message(product, action: 'Registered')
-        log.info "#{action} #{product.identifier} #{product.version} #{product.arch}"
+        log.info "\n#{action} #{product.identifier} #{product.version} #{product.arch}".bold
         log.info "Rooted at: #{@config.filesystem_root}" if @config.filesystem_root
         log.info "To server: #{@config.url}" if @config.url
         log.info "Using E-Mail: #{@config.email}" if @config.email
+        puts "==========\n"
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: The double separator is a bit clunky between modules, can it only be outputted after the last module?

File: lib/suse/connect/client.rb [hunk 5]
```diff
@@ -60,8 +66,9 @@ def deregister!
             deregister_product(product)
           end
           @api.deregister(system_auth)
+          log.info 'Cleaning up ...'
           System.cleanup!
-          log.info 'Successfully deregistered system.'
+          log.info "Successfully deregistered system.\n".red.bold
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: IMO red is for errors, seeing "successfully" in red gives me cognitive dissonance. 

Also see all the error cases for register/deregister that are handled in SUSE::Connect::Cli::execute! - these could use red.

File: lib/suse/connect/client.rb (hunk 4 above)
Comment (fschueller) [reply lib/suse/connect/client.rb]: I used it as a visual separator between the different products, but I can add this only to the final message (or get rid of the separators entirely and only use newlines?)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/158906/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged a2894a92fe6ec853c1555e49d89c1cc0dc4c8401 into 529b91ef340e31070676604ed99b312a588dfedb)

File: lib/suse/connect/client.rb (hunk 2 above)
Comment (fschueller) [reply lib/suse/connect/client.rb]: Not blessed, just invented :angel:

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/158907/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 889231611aad552209b533a6cf8a4822c696bf00 into 529b91ef340e31070676604ed99b312a588dfedb)

File: lib/suse/connect/client.rb (hunk 4 above)
Comment (wstephenson) [reply lib/suse/connect/client.rb]: One separator between products is fine, but I got two when registering SLE15: 
```
91e3485c3993:/tmp/connect # bin/SUSEConnect -d
==========
Deactivating sle-module-server-applications 15 x86_64 ...
-> Removing service from system ...
-> Removing release package ...

Deregistered sle-module-server-applications 15 x86_64
To server: https://scc.suse.com
==========
==========
Deactivating sle-module-basesystem 15 x86_64 ...
-> Removing service from system ...
-> Removing release package ...

Deregistered sle-module-basesystem 15 x86_64
To server: https://scc.suse.com
==========
Cleaning up ...
Successfully deregistered system.
```

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/158923/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 88ca5d3e1572fa6d69822897848f92ee965c457a into 529b91ef340e31070676604ed99b312a588dfedb)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/158924/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 47f2eab049d6d596de929d4eba7882ceeba515ee into 529b91ef340e31070676604ed99b312a588dfedb)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/158927/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged 7fe0b30f29d325a9b53fc1e1f349f8823eec1e5d into 529b91ef340e31070676604ed99b312a588dfedb)

File: lib/suse/connect/client.rb (hunk 1 above)
Comment (fschueller) [reply lib/suse/connect/client.rb]: I understand the issue with the translations, but how does passing a symbol clean this up specifically?

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/158935/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 195616847350225197d0b1bb18ed2ce84160fb23 into 529b91ef340e31070676604ed99b312a588dfedb)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/158948/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 272a706a007883844ab8954b9a865d861ce68903 into 827a32f2b8c50c84043261b1512a93d5675adcee)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/158951/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 250ee7543318b068d5b1e06e98a7a5af2c834bbd into 827a32f2b8c50c84043261b1512a93d5675adcee)

File: lib/suse/connect/client.rb [hunk 6]
```diff
@@ -31,17 +31,22 @@ def register!
...
+        log.info '-> Adding service to system ...'
         System.add_service(service, !@config.no_zypper_refs)
-        Zypper.install_release_package(product.identifier) if install_release_package
+        if install_release_package
+          log.info '-> Installing release package ...'
+          Zypper.install_release_package(product.identifier)
+        end
 
-        print_success_message(product)
+        print_success_message(product, action: 'register')
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: print_success_message(product, :register)

File: lib/suse/connect/client.rb [hunk 7]
```diff
@@ -213,10 +221,12 @@ def register_product_tree(product)
 
       def deregister_product(product)
         raise BaseProductDeactivationError if product == Zypper.base_product
+        log.info "Deactivating #{product.identifier} #{product.version} #{product.arch} ..."
         service = deactivate_product product
         remove_or_refresh_service(service)
+        log.info '-> Removing release package ...'
         Zypper.remove_release_package product.identifier
-        print_success_message product, action: 'Deregistered'
+        print_success_message product, action: 'deregister'
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: print_success_message product, :deregister

File: lib/suse/connect/client.rb [hunk 8]
```diff
@@ -240,17 +250,25 @@ def registered?
...
           Zypper.refresh_all_services
         else
+          log.info '-> Removing service from system ...'
           System.remove_service service
         end
       end
 
-      def print_success_message(product, action: 'Registered')
-        log.info "#{action} #{product.identifier} #{product.version} #{product.arch}"
+      def print_success_message(product, params)
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: def print_success_message(product, action)
   if action == :register

Comment (wstephenson) [reply lib/suse/connect/client.rb]: I think this is all that is required here, clearer than passing a hash with only entry.

Comment (wstephenson) [reply lib/suse/connect/client.rb]: I couldn't reproduce the problem you described with the simple :register/:deregister parameter.

File: lib/suse/connect/client.rb [hunk 9]
```diff
@@ -94,6 +101,7 @@ def announce_system(distro_target = nil, instance_data_file = nil)
       # Re-send the system's hardware details on SCC
       #
       def update_system(distro_target = nil, instance_data_file = nil)
+        log.info "Update system details on SCC/Registration proxy ...\n".bold
```
Comment (wstephenson) [inline lib/suse/connect/client.rb]: Suggest making the voice of this line match the rest of the output - "Updating"/"Updated" as you prefer

File: lib/suse/connect/client.rb (hunk 8 above)
Comment (fschueller) [reply lib/suse/connect/client.rb]: Whuut...okay. I will check that again.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/159815/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp3. (Merged ba6b7185e70a10b0e9fef81e42a007d311e7780a into 827a32f2b8c50c84043261b1512a93d5675adcee)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/159817/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 7ba250117cd82fb812170cb117e081fe52be18c3 into 827a32f2b8c50c84043261b1512a93d5675adcee)

**Pull request merged**